### PR TITLE
[SofaHelper] Load image using stb

### PR DESCRIPTION
**Pull request by fredroy**
Very useful collection of  header-only utilities : https://github.com/nothings/stb

This PR uses **stb_image** and **stb_image_write** to read/write jpeg,png,etc. images directly without any reliance on libpng or libjpeg.
Practically, This PR almost depracates the use of CImgPlugin (at least as an image loader). 
The only thing stb (compared to cimg) is that it cannot load is the TIFF format...

It allows us to :
 - not depend on any external library anymore (libpng, libjpeg)
 - not need anymore to load the CimgPlugin to get images in a scene (which was a bit weird...)

**stb** licence is either MIT or even public domain (!)

Some questions about this PR:
 - should we fetch stb at cmake configure time ? IMO no because it would mean we really need an Internet connection to compile the core of SOFA
 - should we put all the files from the stb repo ? IMO no, the other files wont be used for sure. That's why the stb dir contains only the two files we want for this PR.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
That's a very popular library. I agree with this move.

I don't understand in your PR how it replaces the need for CImgPlugin. Probably, because I don't know how images loading is managed.

**Comment by fredroy** [comment]
> That's a very popular library. I agree with this move.
> 
17.5k ⭐ 😗 (one day, SOFA, one day...)

but yes glfw+imgui+stb is a well-known triplet 

> I don't understand in your PR how it replaces the need for CImgPlugin. Probably, because I don't know how images loading is managed.

Image loading is managed through a factory ("png" -> use this, "bmp" -> use that, etc) and this factory is populated at run-time. 
Before this PR, it was done by CImgPlugin (cimg was used because it can manage several file formats). But it was decided to be optional for some reason, so it was a plugin (and not a "hard" component in the framework).
Now that stb does not depend on any external library, no hassle with cmake/deps/dlls so thats why I put it in SofaHelper.

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by damienmarchal** [comment]
I was afraid of having one more header-only-library, but @fredroy used it the right way as it hide the library in a single translation unit. 
So :+1:

File: SofaKernel/modules/SofaHelper/CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
-# DEPENDENCIES
-# sofa_find_package(Sofa.Type REQUIRED)
-
 # Eigen (header only)
 sofa_find_package(Eigen3 REQUIRED)
 # Json (header only) needed by AdvancedTimer
 sofa_find_package(Json 3.1.2 REQUIRED BOTH_SCOPES)
+# STB (header only)
+sofa_find_package(STB REQUIRED BOTH_SCOPES)
```
Comment (guparan) [inline SofaKernel/modules/SofaHelper/CMakeLists.txt]: BOTH_SCOPES is not needed (same remark for Json above).

**Review by guparan — CHANGES_REQUESTED** [review]

**Review by guparan — APPROVED** [review]

**Pull request merged**